HLTrigger/Timer/plugins/FastTimerService.cc

```diff
@@ -1587,11 +1596,21 @@ void FastTimerService::postModuleEventPrefetching(edm::StreamContext const& sc,
 }
 
 void FastTimerService::preEventReadFromSource(edm::StreamContext const& sc, edm::ModuleCallingContext const& mcc) {
-  ignoredSignal(__func__);
+  if (mcc.state() == edm::ModuleCallingContext::State::kPrefetching) {
+    auto& stream = streams_[sc.streamID()];
+    thread().measure_and_accumulate(stream.overhead);
+  }
 }
 
 void FastTimerService::postEventReadFromSource(edm::StreamContext const& sc, edm::ModuleCallingContext const& mcc) {
-  ignoredSignal(__func__);
+  if (mcc.state() == edm::ModuleCallingContext::State::kPrefetching) {
+    edm::ModuleDescription const& md = callgraph_.source();
+    unsigned int id = md.id();
+    auto& stream = streams_[sc.streamID()];
+    auto& module = stream.modules[id];
+
+    thread().measure_and_accumulate(module.total);
```

Comment:
Do I understand correctly that this will account the time spent reading a collection from the `source` to the `source` itself ?

Reply:
yes